spec/parser/options_spec.rb

```diff
@@ -368,6 +380,18 @@ def remaining
       it "must not allow the same hash key to be specified multiple times" do
         expect {parse("--attributes", "name:string", "name:integer")}.to raise_error(Thor::MalformattedArgumentError, "You can't specify 'name' more than once in option '--attributes'; got name:string and name:integer")
       end
+
+      it "accepts value when value 't match validator" do
+        create :attributes => Thor::Option.new("attributes", :type => :hash, :validator => proc { |k,v| /name/ === k && /string/ === v}, :validator_desc => 'description')
+        expect { parse("--attributes", "name:string") }.not_to raise_error
```

Comment:
It is missing tests to arguments_spec like those one.

Reply:
@rafaelfranca I'd love to see this code merged, and have some time available to work on it. I'm not understanding what kind of tests you would like to see.

Reply:
Test for the argument parser that now also accepts validators.

Reply:
@rafaelfranca The argument parser seems contradictory. It accepts an `Array` of `Argument` objects, then assigns those to a `@switches` variable, then later checks if that variable contains a `Hash`. The iteration within the file also seems very strange, and it's difficult to see how the "enum" argument functionality could be working. The argument code seems to need revision and clarification.

Reply:
@rafaelfranca Is this the test that is missing? Perhaps I have misunderstood: https://github.com/benkoshy/thor/commit/b9a735d46f1ae7a3f09af48259377b89c15d24d3

(btw thank you for your open source contributions)